pkgs/misc/tpm2-pkcs11/default.nix

```diff
@@ -0,0 +1,79 @@
+{ stdenv, lib, fetchFromGitHub, writeText
+, pkgconfig, autoreconfHook, autoconf-archive, makeWrapper, patchelf
+, tpm2-tss, tpm2-tools, opensc, openssl, sqlite, python37, glibc, libyaml
+, abrmdSupport ? true, tpm2-abrmd ? null
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tpm2-pkcs11";
+  version = "1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "tpm2-software";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:06kpf730al50xv1q53ahycky3im23ysrqp40libls4k24zxs9ha2";
+  };
+
+  patches = lib.singleton
+    (writeText "${pname}-${version}-configure-ac-version.patch"
+      (builtins.replaceStrings ["VERSION"] [version]
+        (builtins.readFile ./0001-configure-ac-version.patch)));
+
+  # The preConfigure phase doesn't seem to be working here
+  # ./bootstrap MUST be executed as the first step, before all
+  # of the autoreconfHook stuff
+  postPatch = ''
+    ./bootstrap
+  '';
+
+  nativeBuildInputs = [
+    pkgconfig autoreconfHook autoconf-archive makeWrapper patchelf
+  ];
+  buildInputs = [
+    tpm2-tss tpm2-tools opensc openssl sqlite libyaml
+    (python37.withPackages (ps: [ ps.pyyaml ps.cryptography ps.pyasn1-modules ]))
+  ];
+
+  outputs = [ "out" "tools" ];
+  outputBin = "tools";
```

Comment:
The output for binaries should be `bin`. And since with https://github.com/NixOS/nixpkgs/pull/72029#discussion_r372940587 the library will be installed, you should also add a `dev` output to be able to install the library without the other bits.

Reply:
Maybe even add a `lib` output too, to explicitly have libraries in there

Reply:
Currently, only the library is installed in the `out` output, no other bits. The only output containing any executables is in `tools`. I can change this to `bin` for sure. Is a `dev` output required then?

Reply:
`out` currently includes
```
result
├── lib
│  ├── pkgconfig
│  │  └── tpm2-pkcs11.pc
│  ├── libtpm2_pkcs11.la
│  ├── libtpm2_pkcs11.so -> libtpm2_pkcs11.so.0.0.0
│  ├── libtpm2_pkcs11.so.0 -> libtpm2_pkcs11.so.0.0.0
│  └── libtpm2_pkcs11.so.0.0.0
└── nix-support
   └── propagated-build-inputs
```

The pkgconfig and nix-support bits are development parts. I suggest doing this to split those:
```diff
     (python37.withPackages (ps: [ ps.pyyaml ps.cryptography ps.pyasn1-modules ]))
   ];
 
-  outputs = [ "out" "bin" ];
-  outputBin = "bin";
+  outputs = [ "out" "bin" "dev" ];
 
   dontStrip = true;
   dontPatchELF = true;
```

(and no need to assign `outputBin` because the default is `bin` already)

Reply:
Okay, I've integrated the requested changes. Now there are three outputs:

- ```
  nix/store/452rf48l51c6rcybc217pclcyav9jdlx-tpm2-pkcs11-1.0.1
  └── lib
      ├── libtpm2_pkcs11.la
      ├── libtpm2_pkcs11.so -> libtpm2_pkcs11.so.0.0.0
      ├── libtpm2_pkcs11.so.0 -> libtpm2_pkcs11.so.0.0.0
      └── libtpm2_pkcs11.so.0.0.0
  ```
- ```
  /nix/store/57ic3dsidp6lm4g1wn5h1d76xl0dlwqn-tpm2-pkcs11-1.0.1-dev
  ├── lib
  │   └── pkgconfig
  │       └── tpm2-pkcs11.pc
  └── nix-support
      └── propagated-build-inputs
  ```
- ```
  /nix/store/16xlkaw49ayl10zi6wsbkja44qf3mbbf-tpm2-pkcs11-1.0.1-bin
  ├── bin
  │   └── tpm2_ptool
  └── share
      └── tpm2_pkcs11
          ├── tpm2_pkcs11
          │   ├── tpm2_ptool.py
          │   └── [...]
          ├── tpm2_ptool.py
          └── [...]
  ```

Good to know that this is the right(TM) way to do it.